overlord/snapstate/snapstate.go

```diff
@@ -1764,13 +1768,16 @@ func Remove(st *state.State, name string, revision snap.Revision) (*state.TaskSe
 		prev = disconnect
 	}
 
-	if tp, _ := snapst.Type(); tp == snap.TypeApp && removeAll {
-		ts, err := AutomaticSnapshot(st, name)
-		if err == nil {
-			addNext(ts)
-		} else {
-			if err != ErrNothingToDo {
-				return nil, err
+	// 'purge' flag disables automatic snapshot for given remove op
+	if flags == nil || flags.Purge == false {
+		if tp, _ := snapst.Type(); tp == snap.TypeApp && removeAll {
```

Comment:
I know this is not new code but is there a reason to special case application type? Would it be useful to keep gadget state across remodel?

Reply:
Yes, we can consider changing it (a material for discussion). I've briefly touched it with Michael wrt core when the feature was landing, but it wasn't clear if anything but apps need backups. Happy to reconsider.